connect.go

```diff
@@ -306,6 +309,10 @@ func (opts Options) connect() (*Conn, error) {
 	stationUpdatesSubsLock.Lock()
 	defer stationUpdatesSubsLock.Unlock()
 	c.stationUpdatesSubs = make(map[string]*stationUpdateSub)
+
+	stationFunctionsSubsLock.Lock()
```

Comment:
Why should we lock here?

Reply:
changed to a lock at the station level in order to modify the map without interfering producers